src/components/Header/Header.js

```diff
@@ -0,0 +1,34 @@
+import { MDCTopAppBar } from "@material/top-app-bar";
+import headerTemplate from "./Header.html";
+
+export class HeaderComponent {
+  constructor(mountPoint, props = {}) {
+    this.mountPoint = mountPoint;
+    this.props = props;
+  }
+
+  initMaterial() {
+    this.topAppBar = MDCTopAppBar.attachTo(
+      document.querySelector(".main__header")
+    );
+  }
+
+  querySelectors() {
+    this.topAppBar.setScrollTarget(this.props.scrollTarget);
+  }
+
+  addEventListeners() {
+    this.topAppBar.listen("MDCTopAppBar:nav", this.props.onOpen);
+  }
+
+  mount() {
+    this.mountPoint.innerHTML = this.render();
+    this.initMaterial();
```

Comment:
you should first query selectors and the init material. Please fix this in all cases

Reply:
If I write like this https://screencast.com/t/VRT743Q7wPjM then I am getting an error: 
**Uncaught TypeError: Cannot read property 'setScrollTarget' of undefined** and the site is broken.
I think that specifically for this library we need to init material first. Can we?

Reply:
Or we can make an exception for Header.js component only. For other components we can init material after querySelector and it works fine.

Reply:
Your `querySelectors` method does not do search by selector. So, I think your method should have another name and in such case you can call it after calling `this.initMaterial();`.

Reply:
Thank you. You are right.